### DISCUSSION TITLE
Regression in handling tfenv regex in .terraform-version

### DISCUSSION THREAD

**Discussion by mengesb** [jdx/mise#3833] [Troubleshooting and bug reports]
**Describe the bug**

Mise is not properly parsing `.terraform-version` files from tfenv when regex is present. tfenv allows for a `latest:<regex>` syntax, so something like `latest:^1\.[0-9]\+\.[0-9]\+$` ensures a semver compatible result. Unfortunately using something like `latest:^1\.0` will result in version retuns for `-alpha` or `-beta`, etc, type versions, so this ensures a released compatible version is matched.

**To Reproduce**

1. echo 'latest:^1\.[0-9]\+\.[0-9]\+$' > .terraform-version
2. mise install

**Expected behavior**

mise installs correct version terraform

**`mise doctor` output**

```text
version: 2024.12.20 macos-arm64 (2024-12-25)
activated: yes
shims_on_path: no

build_info:
  Target: aarch64-apple-darwin
  Features: DEFAULT, NATIVE_TLS
  Built: Wed, 25 Dec 2024 20:45:03 +0000
  Rust Version: rustc 1.83.0 (90b35a623 2024-11-26) (Homebrew)
  Profile: release

shell:
  /bin/zsh
  zsh 5.9 (arm64-apple-darwin24.0)

dirs:
  cache: ~/Library/Caches/mise
  config: ~/.config/mise
  data: ~/.local/share/mise
  shims: ~/.local/share/mise/shims
  state: ~/.local/state/mise

env_vars:
  MISE_SHELL=zsh

settings:

1 warning found:

1. plugin python overrides a core plugin

1 problem found:

1. failed to load config: error parsing config file: ~/source/-REDACTED-/-REDACTED-/.terraform-version

mise ERROR error parsing config file: ~/source/-REDACTED-/-REDACTED-/.terraform-version                                                                                      
mise ERROR invalid tool version request: latest:^1\.[0-9]\+\.[0-9]\+$
mise ERROR Run with --verbose or MISE_VERBOSE=1 for more information
```

**Additional context**

Looks like regression of #917 or #1727 

```bash
➜  -REDACTED- git:(main) ✗ tfenv use
Switching default version to v1.10.3
Default version file overridden by /Users/-REDACTED-/source/-REDACTED-/-REDACTED-/.terraform-version, changing the default version has no effect
Default version (when not overridden by .terraform-version or TFENV_TERRAFORM_VERSION) is now: 1.10.3
mise ERROR error parsing config file: ~/source/-REDACTED-/-REDACTED-/.terraform-version                                                                                      
mise ERROR invalid tool version request: latest:^1\.[0-9]\+\.[0-9]\+$
mise ERROR Run with --verbose or MISE_VERBOSE=1 for more information
```

**Comment by mengesb** [comment]
Not entirely sure why this is a discussion, since it seems more like a regression of prior functionality.

**Comment by jdx** [comment]
mise has never supported regex or semver parsing of versions. It's possible the asdf plugin had some logic to handle this but mise no longer defaults to using the asdf plugin for terraform. You can still manually use the plugin if you want.

**Comment by mengesb** [comment]
mise use to play nice ... now it isn't since the update, however I can either disable a plugin entirely or not use mise - there's no middle ground which use to be possible I guess
